node/silkworm/db/bitmap.cpp

```diff
@@ -32,72 +33,49 @@ std::optional<uint64_t> seek(const roaring::Roaring64Map& bitmap, uint64_t n) {
     return std::nullopt;
 }
 
-roaring::Roaring64Map cut_left(roaring::Roaring64Map& bm, uint64_t size_limit) {
-    if (bm.getSizeInBytes() <= size_limit) {
-        roaring::Roaring64Map res(
-            roaring::api::roaring_bitmap_from_range(bm.minimum(), bm.maximum() + 1, 1));  // With range
-        res &= bm;
-        res.runOptimize();
-        bm.clear();
-        return res;
-    }
-    auto from{bm.minimum()};
-    auto min_max{bm.maximum() - bm.minimum()};
-
-    // We look for the cutting point
-    uint64_t i = min_max;
-    uint64_t j = 0;
-    while (i < j) {
-        uint64_t h = (i + j) >> 1;
-        roaring::Roaring64Map current_bitmap(
-            roaring::api::roaring_bitmap_from_range(from, from + i + 1, 1));  // With range
-        current_bitmap &= bm;
-        current_bitmap.runOptimize();
-        if (current_bitmap.getSizeInBytes() <= size_limit) {
-            i = h + 1;
-        } else {
-            j = h;
-        }
-    }
-    roaring::Roaring64Map res(roaring::api::roaring_bitmap_from_range(from, from + i, 1));
-    res &= bm;
-    res.runOptimize();
-    for (uint64_t k = from; k <= from + i; k++) {
+static void remove_range_impl(roaring::Roaring& bm, uint32_t min, uint32_t max) {
+    roaring::api::roaring_bitmap_remove_range(&bm.roaring, min, max);
+}
+
+static void remove_range_impl(roaring::Roaring64Map& bm, uint64_t min, uint64_t max) {
```

Comment:
Is this ever used ?

Reply:
Yes, it is used for `Roaring64Map`. CLion diagnostics is wrong here for some reason.

Reply:
I was not referring to Clion rather the fact it seems it has not test coverage.
Sorry for imprecision